src/rgw/driver/rados/rgw_d3n_datacache.h

```diff
@@ -51,14 +51,27 @@ struct D3nCacheAioWriteRequest {
 	int d3n_prepare_libaio_write_op(bufferlist& bl, unsigned int len, std::string oid, std::string cache_location);
 
   ~D3nCacheAioWriteRequest() {
-    ::close(fd);
+		::close(fd);
 		cb->aio_buf = nullptr;
 		free(data);
 		data = nullptr;
 		delete(cb);
   }
 };
 
+struct D3nFileClose {
+  void operator()(FILE* fd) {
+    if (fd == nullptr) {
+      return;
+    }
+    int retval = fclose(fd);
+    if(retval != 0) {
+      ldpp_dout(dpp, 20) << "ERROR: D3nDataCache::fclose file has return error, errno=" << errno << dendl;
```

Comment:
compile err, have to pass `dpp` to this (or lambda as @yuvalif  suggested)
`../src/rgw/driver/rados/rgw_d3n_datacache.h:69:17: error: use of undeclared identifier 'dpp'; did you mean 'dup'?`